### DISCUSSION TITLE
[adonisjs 4.1] Whitelist IP address middleware?

### DISCUSSION THREAD

**Discussion by aprakasa** [adonisjs/core#1436] [Help]
How to whitelist from IP Address by using middleware?
e.g 192.168.0.0/24

**Comment by alainrc2005** [comment]
Create and register a middleware

```js
async handle ({ request, response }, next) {
	const ip = request.ip()
	if (['127.0.0.1','::1','192.168.1.1','etc'].indexOf(ip)===-1) {
		response.status(403).send(''IP not whitelisted')
		return
	}
	await next()
}
```

[Validate an IP Address (with Mask)](https://stackoverflow.com/questions/4209760/validate-an-ip-address-with-mask)

I hope this can help you

**Reply by aprakasa** [reply]
Thanks, that give an idea 💡
